packages/notte-browser/src/notte_browser/window.py

```
@@ -315,18 +315,66 @@ async def snapshot_metadata(self) -> SnapshotMetadata:
+                            f"CDP session detach for {url} failed after {elapsed_ms:.0f}ms: {type(exc).__name__}: {exc}"
+                        )
+                    elif elapsed_ms > 500:
+                        logger.warning(
+                            f"CDP session detach for {url} took {elapsed_ms:.0f}ms (attach={attach_ms:.0f}ms)"
+                        )
+
+                detach_task.add_done_callback(_log_detach)
+                try:
+                    await asyncio.shield(detach_task)
+                except Exception:
+                    pass  # _log_detach handles logging via the done callback
```

Ruff (0.15.11)

[error] `try`-`except`-`pass` detected, consider logging the exception

(S110)